packages/datatrak-web-server/src/app/createApp.ts

```diff
@@ -33,6 +35,7 @@ export function createApp() {
     .attachApiClientToContext(authHandlerProvider)
     .get<UserRequest>('getUser', handleWith(UserRoute))
     .get<SurveysRequest>('surveys', handleWith(SurveysRoute))
+    .get<SurveyRequest>('surveys/:surveyCode', handleWith(SurveyRoute))
```

Comment:
Just in case you didn't know about it, there is a middleware at the bottom of this file which forwards all routes on to the central server if a route is not matched and it seems like the single survey could just be fetched from there. I think it's ok to keep the custom SurveyRoute that you've written since you've already done the work but it's something to keep in mind for future as it can save a bit of boilerplate code.

Reply:
I initially did use that, but because we want to fetch the survey by code, not id, it wasn't returning any results